src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -0,0 +1,410 @@
+      val image = mesos.Image.newBuilder
+
+      im.forcePull.foreach(forcePull => image.setCached(!forcePull))
+
+      im.kind match {
+        case raml.ImageType.Docker =>
+          val docker = mesos.Image.Docker.newBuilder.setName(im.id)
+
+          image.setType(mesos.Image.Type.DOCKER).setDocker(docker)
+        case raml.ImageType.Appc =>
+          // These labels are necessary for AppC images to work.
+          // Given that Docker only works under linux with 64bit,
+          // let's (for now) set these values to reflect that.
+          val labels = mesos.Labels.newBuilder
+            .addAllLabels(
+              Iterable(
+              mesos.Label.newBuilder.setKey("os").setValue("linux").build,
+              mesos.Label.newBuilder.setKey("arch").setValue("amd64").build
+            ).asJava)
+
+          val appc = mesos.Image.Appc.newBuilder.setName(im.id).setLabels(labels)
+
+          image.setType(mesos.Image.Type.APPC).setAppc(appc)
+      }
+
+      val mesosInfo = mesos.ContainerInfo.MesosInfo.newBuilder.setImage(image)
+      containerInfo.setMesos(mesosInfo)
+    }
+
+    containerInfo
+  }
+
+  private[this] def computeHealthCheck(
+    healthCheck: raml.HealthCheck,
+    endpoints: Seq[raml.Endpoint]): mesos.HealthCheck.Builder = {
+    val builder = mesos.HealthCheck.newBuilder
+
+    if (healthCheck.command.isDefined) {
+      builder.setType(mesos.HealthCheck.Type.COMMAND)
+
+      val command = mesos.CommandInfo.newBuilder
+
+      healthCheck.command.get.command match {
+        case raml.ShellCommand(shell) =>
+          command.setShell(true)
+          command.setValue(shell)
+        case raml.ArgvCommand(argv) =>
+          command.setShell(false)
+          if (argv.nonEmpty) {
+            command.setValue(argv.head)
+          }
+
+          if (argv.size > 1) {
+            command.addAllArguments(argv.tail.asJava)
+          }
+      }
+
+      builder.setCommand(command)
+    } else if (healthCheck.http.isDefined) {
+      builder.setType(mesos.HealthCheck.Type.HTTP)
+
+      val http = mesos.HealthCheck.HTTPCheckInfo.newBuilder
+
+      healthCheck.http.get.scheme.foreach(scheme => http.setScheme(scheme.value))
+      healthCheck.http.get.path.foreach(http.setPath)
+
+      builder.setHttp(http)
+    } else if (healthCheck.tcp.isDefined) {
+      builder.setType(mesos.HealthCheck.Type.TCP)
+
+      val tcp = mesos.HealthCheck.TCPCheckInfo.newBuilder
+
+      endpoints.find(_.name == healthCheck.tcp.get.endpoint).foreach{ endpoint =>
+        // TODO: determine if not in "HOST" mode and use the container port instead
+        endpoint.hostPort.foreach(tcp.setPort)
+      }
+
+      builder.setTcp(tcp)
+    }
+
+    builder
+  }
+
+  private[this] def taskContextEnv(
+    container: raml.MesosContainer,
+    version: Timestamp,
+    taskId: Instance.Id): Map[String, String] = {
+    Map(
+      "MESOS_TASK_ID" -> Some(taskId.idString),
+      "MARATHON_APP_ID" -> Some(container.name),
```

Comment:
Yes, MARATHON_APP_ID should be the runSpecId

On Mon, Sep 12, 2016 at 9:10 AM, Jan Schlicht <notifications@github.com>
wrote:

> In src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala
> <https://github.com/mesosphere/marathon/pull/4353#discussion_r78368531>:
>
> > +        endpoint.hostPort.foreach(tcp.setPort)
> > +      }
> > +
> > +      builder.setTcp(tcp)
> > +    }
> > +
> > +    builder
> > +  }
> > +
> > +  private[this] def taskContextEnv(
> > +    container: raml.MesosContainer,
> > +    version: Timestamp,
> > +    taskId: Instance.Id): Map[String, String] = {
> > +    Map(
> > +      "MESOS_TASK_ID" -> Some(taskId.idString),
> > +      "MARATHON_APP_ID" -> Some(container.name),
>
> Should we propagate the runSpecId as well, i.e. keep MARATHON_APP_ID, but
> set it to runSpecId?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4353/files/0a69499a62feb921a20eff42c484a997f004cd01#r78368531>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACPVLOPq9TJ_n19mTJAO1G2V6uoz2775ks5qpU8tgaJpZM4J6bJ5>
> .
>